riot_pal/dut_pyshell.py

```diff
@@ -0,0 +1,258 @@
+#! /usr/bin/env python3
+# Copyright (c) 2019 Kevin Weiss, for HAW Hamburg  <kevin.weiss@haw-hamburg.de>
+#
+# This file is subject to the terms and conditions of the MIT License. See the
+# file LICENSE in the top level directory for more details.
+# SPDX-License-Identifier:    MIT
+
+"""
+This script handles interfacing to the PHiLIP device.  It parses data and
+exposes from the PhilipExtIf class to be run in a shell.
+
+The purpose of this script is allow easy setup and manual usage of PHiLIP.
+
+Usage
+-----
+
+```
+usage: dut_pyshell.py   [-h]
+                        [--loglevel {debug,info,warning,error,fatal,critical}]
+                        [--port PORT]
+                        [--filter-data]
+
+optional arguments:
+  -h, --help            show this help message and exit
+  --loglevel {debug,info,warning,error,fatal,critical}
+                        Python logger log level (default: warning)
+  --port, -p
+                        Specify the serial port
+  --data_only, -do
+                        Filters data from philip responses to only display
+                        what is needed (default: True)
+```
+"""
+import cmd
+from json import dumps
+import logging
+import argparse
+import serial.tools.list_ports
+try:
+    import readline
+except ImportError:
+    readline = None
+try:
+    from .dut_shell import DutShell, RESULT_SUCCESS
+except ImportError:
+    from dut_shell import DutShell, RESULT_SUCCESS
+
+
+class DutPyShell(cmd.Cmd):
+    """Command loop for the PHiLIP interface
+
+    Args:
+        port - Serial port for the PHiLIP, if None connection wizard tries to
+               connent
+        data_only - If true only data prints from command an not the whole
```

Comment:
why not call this `debug` and invert the logic? At least to me that is what other programs do.

Reply:
Heh it should actually be called filter data.
I wanted debug to be a bit more verbose and unstructured.  Without the filter it is very structured and with the filter it is more human readable.

Reply:
mhm, okay maybe I need to look at it again and rethink ...